### PR TITLE
configcore: register virtual config for timezone reading

### PR DESCRIPTION
**Pull request by mvo5**
This commit uses the new virtual configuration mechanism to make
the `snap get core system.timezone` output look at the system
setting instead of at what is stored in the state. It also adds
a test that ensures that writing the timezone will not store it
in the state.

### CODE REVIEW

File: overlord/configstate/configcore/timezone.go [hunk 1]
```diff
@@ -34,6 +36,8 @@ import (
 func init() {
 	// add supported configuration of this module
 	supportedConfigurations["core.system.timezone"] = true
+	// and register it as a virtual config
+	config.RegisterVirtualConfig("core", "system.timezone", getTimezoneFromSystemVC)
```
Comment (pedronis) [inline overlord/configstate/configcore/timezone.go:40]: this is not related to the PR, but as we discussed in a the previous PR, virtual is maybe not the best term, it's about the implementation more than the use case.  Maybe s/Vitual/External/ in a follow up?

**Review by pedronis — APPROVED** [review]
thanks, one comment for a follow-up

File: overlord/configstate/configcore/timezone.go (hunk 1 above)
Comment (mardy) [reply overlord/configstate/configcore/timezone.go:40]: Other options (assuming I got the intention right): dynamic, runtime.

File: overlord/configstate/configcore/timezone.go [hunk 2]
```diff
@@ -95,3 +108,23 @@ func handleTimezoneConfiguration(_ sysconfig.Device, tr config.ConfGetter, opts
...
+func getTimezoneFromSystemVC(key string) (interface{}, error) {
+	return getTimezoneFromSystem()
+}
+
+func getTimezoneFromSystem() (string, error) {
+	// We cannot use "timedatectl show" here because it is only
+	// available on UC20
+	link, err := os.Readlink(filepath.Join(dirs.GlobalRootDir, "/etc/writable/localtime"))
+	// see localtime(5)
+	// "If /etc/localtime is missing, the default "UTC" timezone is used."
```
Comment (mardy) [inline overlord/configstate/configcore/timezone.go:121]: Do we expect to have `/etc/writable/localtime` in all systems? In other words, don't we need to fall back and check also `/etc/localtime`?

(if not, the code is correct, but the comment about `/etc/localtime` could be expanded to clarify that)

File: overlord/configstate/configcore/timezone.go (hunk 1 above)
Comment (mvo5) [reply overlord/configstate/configcore/timezone.go:40]: I added a XXX and will do it in a followup

File: overlord/configstate/configcore/timezone.go (hunk 2 above)
Comment (mvo5) [reply overlord/configstate/configcore/timezone.go:124]: I added a comment - this code will only run on Ubuntu Core systems which all have /etc/writable. Never on classic.

Comment (mardy) [reply overlord/configstate/configcore/timezone.go:124]: Thanks!

**Review by mardy — APPROVED** [review]
LGTM!

**Pull request merged**